azure-pipelines/end-to-end-tests-dir/asset-caching.ps1

```diff
@@ -82,26 +84,118 @@ if (-not ($actual -match $expected)) {
 # Testing asset caching && x-block-orgin promises when --debug is passed (enabled)
 Refresh-TestRoot
 $actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("install", "vcpkg-internal-e2e-test-port", "--overlay-ports=$PSScriptRoot/../e2e-ports", "--x-asset-sources=x-azurl,file://$AssetCache,,readwrite;x-block-origin", "--downloads-root=$DownloadsRoot", "--debug"))
-$actual = $actual -replace "`r`n", "`n"
-
-# Define the regex pattern that accounts for multiline input
-$expectedPattern = "(?s)" +
-                   ".*\[DEBUG\] External asset downloads are blocked \(x-block-origin is enabled\)\.\.\.?" +
-                   ".*\[DEBUG\] Asset caching is enabled\..*"
-
-if (-not ($actual -match $expectedPattern)) {
+if (-not ($actual.Contains("[DEBUG] External asset downloads are blocked (x-block-origin is enabled)") -and $actual.Contains("[DEBUG] Asset caching is enabled."))) {
     throw "Failure: couldn't find expected debug promises (asset caching enabled + x-block-origin enabled)"
 }
 
 # Testing asset caching && x-block-orgin promises when --debug is passed (disabled)
 Refresh-TestRoot
 $actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("install", "vcpkg-internal-e2e-test-port", "--overlay-ports=$PSScriptRoot/../e2e-ports", "--x-asset-sources=clear", "--downloads-root=$DownloadsRoot", "--debug"))
+if (-not ($actual.Contains("[DEBUG] External asset downloads are allowed (x-block-origin is disabled)") -and $actual.Contains("[DEBUG] Asset cache is not configured"))) {
+    throw "Failure: couldn't find expected debug promises (asset caching disabled + x-block-origin disabled)"
+}
+
+# azurl (no), x-block-origin (no), asset-cache (n/a), download (fail)
+# Expected: Download failure message, nothing about asset caching
+Refresh-TestRoot
+$actual = Run-VcpkgAndCaptureOutput -TestArgs ($commonArgs + @("x-download", "$downloadsRoot/example3.html", "--sha512", "d06b93c883f8126a04589937a884032df031b05518eed9d433efb6447834df2596aebd500d69b8283e5702d988ed49655ae654c1683c7a4ae58bfa6b92f2b73a", "--url", "https://localhost:1234/foobar.html"))
+if (-not ($actual.Contains("error: https://localhost:1234/foobar.html: curl failed to download with exit code 7"))) {
+    throw "Failure: azurl (no), x-block-origin (no), asset-cache (n/a), download (fail)"
+}
+
+#azurl (no), x-block-origin (no), asset-cache (n/a), download (sha-mismatch)
+#Expected: Download message with the "you might need to configure a proxy" message and with expected/actual sha
```

Comment:
```
Downloading example3.html
Failed to download example3.html.
If you are using a proxy, please ensure your proxy settings are correct.
Possible causes are:
1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable to `https//address:port`.
This is not correct, because `https://` prefix claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr, etc...) is an HTTP proxy.
Try setting `http://address:port` to both HTTP_PROXY and HTTPS_PROXY instead.
2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings set by your proxy software. See, https://github.com/microsoft/vcpkg-tool/pull/77
The value set by your proxy might be wrong, or have same `https://` prefix issue.
3. Your proxy's remote server is our of service.
If you've tried directly download the link, and believe this is not a temporay download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
to report this upstream download server failure.
error: https://localhost:1234/foobar.html: curl failed to download with exit code 7
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

curl: (7) Failed to connect to localhost port 1234 after 0 ms: Couldn't connect to server
```